### ISSUE TITLE
User creating multiple servers instead of one

### ISSUE THREAD

**Issue by Darrakas** [ControlPanel-gg/dashboard#605]
**Describe the bug 🐛**

I found that when creating server, by pressing "Create Server" button multiple times before the page refresh, it creates as many servers as you click the button (before server throws error).

**To Reproduce**
1. Go to 'https://panel.domain.com/servers'
2. Click on 'Create Server'
3. Choose server configuration
4. Repeatedly click "Create Server" button before page refresh.

**Expected behavior**
To create only one server, no matter how many times you press the button.

**Screenshots**
https://i.imgur.com/WJSi6aH.gif

**Desktop (please complete the following information):**
 - OS: Windows 11
 - Browser: Chrome/Mozilla

**Additional context**

I guess there should be some sort of rate limit that no matter if you either accidentally or intentionally press "Create Server" button repeatedly, it still should only create one server unless you have configured otherwise (?).

Found this by accidentally clicking the button twice and was wondering why it created two servers instead of one. I managed to create 4 when rapidly pressing the button, before 500 error appeared. So I was thinking, if user accidentally pressed the button more than once, or if user is impatient/there is latency, this might cause problems. Main problem is that every created server does charge the first hour and you won't get credits back by deleting "unwanted" servers.

Already asked this in Discord and thought I probably should make issue in here about this.

**Comment by Wouter0100** [comment]
Normally this is solved by disabling the button after the first click. Very easy to do with the onclick event. Can even be done in-line.l

**Comment by 1day2die** [comment]
yeah i tried it with your approach. sadly it didnt work. If you can fix it thatd be great.

The code is written with alpinejs which is a pain in the ass and using onClick didnt work. But go ahead! :)

**Comment by BolverBlitz** [comment]
Ok, this button does very diffrent things depending on the browser used.

First of all, this bug only exists in chromium browsers (Chrome, Edge). Firefox its not reproducable.

Second, the disable Button JS works. BUT again, very diffrent.

Chrome: Submit Event dosn´t get send. Basicly servercreation is impossible.
Firefox: Submit Event is send, but rejected with "The product field is required."

The workaround i found, fixing this issue in both browsers is to just hide the button because something you can´t see can´t be pressed. But this looks a bit silly...

```html
<div x-data="{ open: true }">
    <button type="submit" x-model="selectedProduct" name="product"
    :disabled="product.minimum_credits > user.credits||product.doesNotFit == true"
    :class="product.minimum_credits > user.credits ? 'disabled' : ''"
    x-on:click="open = ! open"
    x-show="open"
    class="btn btn-primary btn-block mt-2" @click="setProduct(product.id)"
    x-text=" product.doesNotFit == true? '{{ __("Server cant fit on this Node") }}' : (product.minimum_credits > user.credits ? '{{ __('Not enough') }} {{ CREDITS_DISPLAY_NAME }}!' : '{{ __('Create server') }}')">
    </button>
</div>
```

**Comment by KroZen-Dev** [comment]
Hello!

The issue described in #605 ("User creating multiple servers instead of one") is still occurring for me:

If I click the "Create Server" button multiple times quickly before the page refreshes, a server is created for each click. For example, clicking the button 5 times results in 5 servers being created. If I click once, one server is created as expected, but any additional rapid clicks create multiple servers.

Please check again and make sure the bug is fully fixed. I am ready to provide additional information or a video if needed.

Thank you!
